File: segundos 4 lenguajes/js/JugadorCurador.js
export class JugadorCurador {
    constructor(nombre = "Curador", nivel = 1, copia = null) {
        if (copia) {
            this.nombre = copia.nombre;
            this.nivel = copia.nivel;
        } else {
            this.nombre = nombre;
            this.nivel = nivel;
        }
    }

    curar() {
        throw new Error("Método curar() debe implementarse.");
    }
}

File: segundos 4 lenguajes/Java/JugadorCurador.java
package Java;
abstract class JugadorCurador {
    String nombre;
    int nivel;

    public JugadorCurador() {}

    public JugadorCurador(String nombre, int nivel) {
        this.nombre = nombre;
        this.nivel = nivel;
    }

    public JugadorCurador(String nombre) {
        this.nombre = nombre;
        this.nivel = 1;
    }

    public JugadorCurador(JugadorCurador j) {
        nombre = j.nombre;
        nivel = j.nivel;
    }

    abstract void curar();
}

